chain/client/src/client.rs

```diff
@@ -919,6 +919,11 @@ impl Client {
         let gas_used = chunk_extra.gas_used();
         #[cfg(feature = "test_features")]
         let gas_used = if self.produce_invalid_chunks { gas_used + 1 } else { gas_used };
+
+        // The congestion info is set to default if it is not present. If the
+        // congestion control feature is not enabled the congestion info will be
+        // stripped from the chunk header anyway. In the first chunk where
+        // feature is enabled the header will contain the default congestion info.
```

Comment:
This is a bit mind boggling... But thinking it through, at least I was able to convince myself it is 100% correct. :)

But I wonder, looking at this code in isolation, wouldn't it make more sense to make the parameter in `create_encoded_shard_chunk` an Option and then unwrap inside `EncodedShardChunk::new()` when we try to encode a chunk with congestion control enabled without valid congestion info?

Completely optional refactor, of course. It's matter of preferences and also depends on how the refactor affects the code in other places.

Reply:
yeah it broke me a little
If I have some time I'll do the refactor but I want to write the nayduck tests first.